Update code examples to match lessons and clean up formatting/typos in lessons where needed (#112)
Ada-Developers-Academy/ada-precourse-v2: intro-to-python/helper-functions.md

<br>

<details>
<summary>Expand for a breakdown of the <code>sub_total</code> variable assignment in <code>display_order_summary</code>.</summary>

  1. `sub_total` is set up to receive a variable assignment, so Python looks at the code to the right of the assignment, where it sees `calculate_subtotal(item_prices)`.
  1. `calculate_subtotal` runs and returns a value of 23.00 when finished.

>   1. The expression result, 23.00, is assigned to `subtotal`. 

Reviewer: ```suggestion
  1. The expression result, 23.00, is assigned to `sub_total`. 
```

# before right-hand side evaluation
sub_total = calculate_subtotal(item_prices)

# after right-hand side evaluation `sub_total` references the value 23.00

> sub_total == 23.00 

Author: I added more details to the comment and use `==` to try and express that sub_total will reference the value 23, and not that we are assigning sub_total 23. What do you think?

Reviewer: I'm getting a little stuck on this myself, because I feel like theses lines of code could easily be taken the wrong way. I tried something kind of between the original and your updates below, is that clearer or kinda the same?  

```suggestion
# after right-hand side evaluation, `calculate_subtotal(item_prices)` returns 23.00
# `sub_total` is assigned that return value and references the value 23.00
sub_total = 23.00 
```

Author: I like how you re-worded the comment on line 25, but it still seems like we're assigning something to the variable but maybe that's just how I perceive it. 

Is this too out of left field:

```
# before right-hand side evaluation
sub_total = calculate_subtotal(item_prices)

# after right-hand side evaluation, `calculate_subtotal(item_prices)` returns 23.00
print(sub_total) #  => 23.00 
```

Reviewer: I like this better to avoid thinking we're doing a second assignment with the `sub_total` variable.

print(total_cost) # => 8300.0
```

> Notice that the expression for calculating the total cost is quite long and a bit unclear. Why is each computation within the mathematical expression necessary? One way to add clarity is to split up the expression into multiple lines, which will make your file longer, but there are times that we should prioritize readability over writing less code or shorter code. 

Reviewer: I wanted to change the tone of this to push on writing readable code over short code a little bit harder, what do you think?

```suggestion
Notice that the expression for calculating the total cost is quite long and a bit unclear. Why is each computation within the mathematical expression necessary? One way to add clarity is to split up the expression into multiple lines. Doing so will make your file longer, but that's okay, most times we should prioritize readability over writing the shortest amount of code possible. 
```
